crc32_fold.h

```diff
@@ -5,8 +5,12 @@
 #ifndef CRC32_FOLD_H_
 #define CRC32_FOLD_H_
 
+#define CRC32_FOLD_BUFFER_SIZE (16 * 5) + 16
+/* (sizeof(__m128i) * (4 folds & 1 partial fold)) + extra space for 16-byte alignment */
+
 typedef struct crc32_fold_s {
-    uint32_t ALIGNED_(16) fold[4 * 5];
+    uint8_t *fold;
```

Comment:
This is bad, pointer is only ever 64 bits (4 or 8 bytes), but we are expecting 128-bits (16 bytes) to align properly. As such this assumes we are always misaligned.

Reply:
Ok I have fixed it by making `crc32_fold_s` in inflate and deflate still have the `ALIGNED_`.

Reply:
And moved buffer to be the first in the `crc32_fold_s` structure.